app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java

```diff
@@ -287,6 +288,11 @@ public void onPause() {
         _appSettings.addRecentFile(_document.getFile());
         _appSettings.setDocumentPreviewState(_document.getPath(), _isPreviewVisible);
         _appSettings.setLastEditPosition(_document.getPath(), _hlEditor.getSelectionStart());
+
+        if(_document.getPath().equals(_appSettings.getTodoFile().getPath()))
```

Comment:
I would make an appSettings function called isTodoFile or something

I would also make sure this does not slow down opening / closing the todo file

Reply:
I didn't find any other usage like this (""istodo file check"), so suggest to keep it like that. Just bloats a new method which single usage then.

Opening we should be fine by usual file changed detection rules, as long widget writes when not editing anymore.

Closing not sure maybe the call can be called in a new started thread

Reply:
Ok. Fair.

Either way one should absolute paths

Reply:
Roughly right or do you have something in mind which needs special attention?

guess the most important part is that the widget actually writes & reads filesystem at the right times and not resets it to older state